base_layer/core/src/transactions/transaction_components/output_features.rs

```diff
@@ -115,27 +132,88 @@ impl OutputFeatures {
         )
     }
 
-    pub fn create_coinbase(maturity_height: u64) -> OutputFeatures {
+    pub fn create_coinbase(maturity_height: u64, recovery_byte: u8) -> OutputFeatures {
         OutputFeatures {
             flags: OutputFlags::COINBASE_OUTPUT,
             maturity: maturity_height,
+            recovery_byte,
             ..Default::default()
         }
     }
 
-    /// Create an `OutputFeatures` with the given maturity and all other values at their default setting
-    pub fn with_maturity(maturity: u64) -> OutputFeatures {
-        OutputFeatures {
-            maturity,
-            ..Default::default()
+    /// Helper function to create a unique recovery byte based on the commitment and a private recovery byte key,
+    /// with the value '0' never obtainable as it is being reserved as a default value.
+    pub fn create_unique_recovery_byte(commitment: &PedersenCommitment, rewind_data: Option<&RewindData>) -> u8 {
+        let commitment_bytes = commitment.as_bytes();
+        let recovery_key_bytes = if let Some(data) = rewind_data {
+            data.recovery_byte_key.as_bytes()
+        } else {
+            &[]
+        };
+        const RECOVERY_BYTE_SIZE: usize = 1;
+        let blake2_hasher = VarBlake2b::new(RECOVERY_BYTE_SIZE)
+            .expect("Should be able to create blake2 hasher; will only panic if output size is 0 or greater than 64");
+        let mut hash = [OutputFeatures::RECOVERY_BYTE_DEFAULT; RECOVERY_BYTE_SIZE];
+        let mut salt = 0u64;
+        loop {
+            blake2_hasher
+                .clone()
+                .chain(commitment_bytes)
+                .chain(recovery_key_bytes)
+                .chain(b"hash my recovery byte")
+                .chain(salt.to_le_bytes().as_slice())
+                .finalize_variable(|res| hash.copy_from_slice(res));
+            if hash[0] == OutputFeatures::RECOVERY_BYTE_DEFAULT {
+                salt += 1;
+            } else {
+                break;
+            }
         }
+        hash[0]
     }
 
-    pub fn custom(flags: OutputFlags, metadata: Vec<u8>) -> OutputFeatures {
-        Self {
-            flags,
-            metadata,
-            ..Default::default()
+    /// Helper function to update the unique recovery byte if required and return updated output features
+    pub fn update_recovery_byte_if_required(
+        commitment: &PedersenCommitment,
+        rewind_data: Option<&RewindData>,
+        features: &OutputFeatures,
```

Comment:
I would expect this signature to take a mut, since it implies that the intention was to change the features?
```suggestion
        features: &mut OutputFeatures,
```

A better signature would be `update_recovery_byte(&self, commitment: .., rewind_data:...) -> ()`